resources/Microsoft.NET.SDK.ToolInstaller/Microsoft.NET.SDK.ToolInstaller.psm1

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+using namespace System.Collections.Generic
+
+$ErrorActionPreference = "Stop"
+$PSNativeCommandUseErrorActionPreference = $true
+Set-StrictMode -Version Latest
+
+#region Functions
+function Get-DotNetPath {
+    if ($IsWindows) {
+        $dotNetPath = "$env:ProgramFiles\dotnet\dotnet.exe"
+        if (-not (Test-Path $dotNetPath)) {
+            $dotNetPath = "${env:ProgramFiles(x86)}\dotnet\dotnet.exe"
+            if (-not (Test-Path $dotNetPath)) {
+                throw "dotnet.exe not found in Program Files or Program Files (x86)"
+            }
+        }
+    } elseif ($IsMacOS) {
+        $dotNetPath = "/usr/local/share/dotnet/dotnet"
+        if (-not (Test-Path $dotNetPath)) {
+            $dotNetPath = "/usr/local/bin/dotnet"
+            if (-not (Test-Path $dotNetPath)) {
+                throw "dotnet not found in /usr/local/share/dotnet or /usr/local/bin"
+            }
+        }
+    } elseif ($IsLinux) {
+        $dotNetPath = "/usr/share/dotnet/dotnet"
+        if (-not (Test-Path $dotNetPath)) {
+            $dotNetPath = "/usr/bin/dotnet"
+            if (-not (Test-Path $dotNetPath)) {
+                throw "dotnet not found in /usr/share/dotnet or /usr/bin"
+            }
+        }
+    } else {
+        throw "Unsupported operating system"
+    }
+
+    Write-Verbose -Message "'dotnet' found at $dotNetPath"
+    return $dotNetPath
+}
+
+# TODO: when https://github.com/dotnet/sdk/pull/37394 is documented and version is released with option simple use --format=JSON
```

Comment:
The linked PR is fixed in 9 it looks like. Were you waiting for 9 to be the default to replace the below code?

Reply:
You are right, I forgot I couldn't install the latest preview from the official website. Everytime when I tried downloading, it gives me 0 bytes. When I used `winget install Microsoft.DotNet.SDK.Preview`, it did. I guess it installs from a different location.

Eventually, I guess we can move the logic to this:

![image](https://github.com/user-attachments/assets/119ce361-435a-40c2-ac77-30572d57e50d)

But we should also be able to support older versions of the SDK?